src/dpo/src/Optdp.cpp

```diff
@@ -0,0 +1,855 @@
+  // optimizations aimed at wirelength improvement.  The last
+  // call to the random improver can be set to consider things
+  // like density, displacement, etc. in addition to wirelength.
+
+  dpo::DetailedParams dtParams;
+  dtParams.m_script = "";
+  // Maximum independent set matching.
+  dtParams.m_script += "mis -p 10 -t 0.005";
+  dtParams.m_script += ";";
+  // Global swaps - commented out right now.  caused padding violation???
+  //dtParams.m_script += "gs -p 10 -t 0.005";
+  //dtParams.m_script += ";";
+  // Vertical swaps - commented out right now.  caused padding violation???
+  //dtParams.m_script += "vs -p 10 -t 0.005";
+  //dtParams.m_script += ";";
+  // Small reordering.
+  dtParams.m_script += "ro -p 10 -t 0.005";
+  dtParams.m_script += ";";
+  // Random moves and swaps with hpwl as a cost function.
+  // Commented out moves generated with global and vertical swaps 
+  // since they seem to be causing padding violations.
+  dtParams.m_script +=
+       "default -p 5 -f 20 -gen rng -obj hpwl -cost (hpwl)";
+  dpo::Detailed dt(dtParams);
+  dt.improve(mgr);
+
+  // Write solution back.
+  updateDbInstLocations();
+
+  hpwlAfter_ = opendp->hpwl();
+
+  double dbu_micron = db_->getTech()->getDbUnitsPerMicron();
+
+  // Statistics.
+  logger_->report("Detailed Improvement Results");
+  logger_->report("------------------------------------------");
+  logger_->report("Original HPWL         {:10.1f} u", (hpwlBefore_/dbu_micron));
+  logger_->report("Final HPWL            {:10.1f} u", (hpwlAfter_/dbu_micron));
+  double hpwl_delta = (hpwlBefore_ == 0.0)
+    ? 0.0
+    : ((double)(hpwlAfter_ - hpwlBefore_) / (double)hpwlBefore_) * 100.;
+  logger_->report("Delta HPWL            {:10.1f} %", hpwl_delta);
+  logger_->report("");
+
+  // Cleanup.
+  if (nw_) delete nw_;
+  if (arch_) delete arch_;
+  if (rt_) delete rt_;
+  nw_ = 0;
+  arch_ = 0;
+  rt_ = 0;
+}
+////////////////////////////////////////////////////////////////
+void Optdp::import() {
+  logger_->report("Importing netlist into detailed improver.");
+
+  nw_ = new Network;
+  arch_ = new Architecture;
+  rt_ = new RoutingParams;
+
+  initEdgeTypes(); // Does nothing right now.
+  initCellSpacingTable(); // Does nothing right now.
+  createLayerMap(); // Does nothing right now.
+  createNdrMap(); // Does nothing right now.
+  setupMasterPowers();  // Call prior to network and architecture creation.
+  createNetwork(); // Create network; _MUST_ do before architecture.
+  createArchitecture(); // Create architecture.
+  createRouteGrid(); // Bad name.  Holds routing info for placement, but not used right now.
+  initPadding(); // Need to do after network creation.
+  setUpNdrRules(); // Does nothing right now.
+  setUpPlacementRegions(); // Regions.
+}
+////////////////////////////////////////////////////////////////
+void Optdp::updateDbInstLocations() {
+  std::unordered_map<odb::dbInst*, Node*>::iterator it_n;
```

Comment:
Its best to declare at the point of usage for clarity.  In this case I would delete this line and just use further down:

auto it_n = instMap_.find(inst);

If you prefer explicit types:

std::unordered_map<odb::dbInst*, Node*>::iterator it_n = instMap_.find(inst);

There are more of the same about declaration at the point of use - I won't keep mentioning it but it applies throughout.